### PR TITLE
Tao 1649 saving data of repeated deliveries

### PR DESCRIPTION
**Pull request by hutnikau**
I know there are a lot of changes. But i could not do it in another way. Please sorry for that. Please do not hesitate to ask me any questions in any time!

### CODE REVIEW

File: manifest.php [hunk 1]
```diff
@@ -23,7 +23,7 @@
     'label' => 'Delivery schedule',
     'description' => 'An extension is used to visualize the delivery schedule in the form of a calendar.',
     'license' => 'GPL-2.0',
-    'version' => '0.1.2',
+    'version' => '0.1.4',
```
Comment (ssipasseuth) [inline manifest.php]: why increasing by 2 version?

Comment (hutnikau) [inline manifest.php]: fixed.

**Comment by ssipasseuth** [comment]
I still have a warning "Missing or invalid parameter classUri " when I try to edit any delivery after running the update script. On a fresh install, I even have an error 500: Service "taoDeliverySchedule/DeliveryGroupsService" not found

**Comment by hutnikau** [comment]
I forgot register services during installation. Now it should work. Please try it again.

**Comment by hutnikau** [comment]
@ssipasseuth, i have fixed request parameters fetching. Please test it.

**Comment by ssipasseuth** [comment]
Yes, I can save it now.
However it appears that every repeated delivery share the same configuration sometimes. When I exclude a test taker from one reapeated delivery, ti is also excluded from the other (but it does not exclude it from the main delivery). When I add a group to a repeated delivery, it only adds to the group only to that repeated delivery, not for the others it seems.

**Comment by ssipasseuth** [comment]
@hutnikau, I have other urgent tasks and PR to do so I am assigning this to @antoinerobin.
@antoinerobin thanks so much !

**Comment by antoinerobin** [comment]
When I register a new repeated delivery and try to modify a sub delivery I have the error message ```Delivery has no start and end date.```

File: controller/CalendarApi.php [hunk 2]
```diff
@@ -291,9 +291,32 @@ private function formatDate($date) {
...
+    /**
+     * Get delivery by request params.
+
+     * @throws \tao_models_classes_MissingRequestParameterException
+     * @returns \core_kernel_classes_Class Delivery instance.
+     */
+    private function getDelivery() {
+        $params = $this->getRequestParams();
+
+        if (empty($params['classUri'])) {
```
Comment (antoinerobin) [inline controller/CalendarApi.php]: Why do you check this parameter whereas you don't use it to get delivery ?

File: scripts/update/Updater.php [hunk 3]
```diff
@@ -55,7 +57,30 @@ public function update($initialVersion) {
 
         if ($currentVersion == '0.1.1') {
             OntologyUpdater::syncModels();
-            $current = '0.1.2';
+            $currentVersion = '0.1.2';
+        }
+
+        if ($currentVersion === '0.1.2') {
+            OntologyUpdater::syncModels();
```
Comment (antoinerobin) [inline scripts/update/Updater.php]: Why do you have to do this ? I can't see where you modify the ontology

**Pull request merged**